java/core/src/main/java/com/lancedb/lance/ipc/Query.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance.ipc;
+
+import org.apache.arrow.util.Preconditions;
+
+import java.util.Optional;
+
+public class Query {
+  public enum DistanceType {
+    L2,
+    Cosine,
+    Dot,
+    Hamming;
+  }
+
+  private final String column;
```

Comment:
Lets make this `columns` to leave room for combined index

Reply:
This one is 100% mimic the https://github.com/lancedb/lance/blob/eb87bfa117c95419b26fe7c929290d439a8206ad/rust/lance-index/src/vector.rs#L57
Once rust API supports columns, we can add it here